keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -1065,6 +1065,51 @@ Details to consider:
 
 #### CEL Expression Composition
 
+##### Policy Predicates
+
+The match criteria in bindings are not expected to be able to cover all possible
+ways users may want to scope their policies. For example, there is no way to
+match off of kind, only resource. To provide extensibility for the match critiera
+without requiring modifying every validation rule individually, a global predicate
+system is needed. These predicates contain CEL statements that must be satisfied, otherwise
+the policy will be ignored. In order to keep bindings language-agnostic and to support
+singleton policies, the logic should live in the policy definition resource. To enable
+customization per-binding, the CEL statements should have access to the parameter resource.
+
+Here is an example of a policy definition using policy predicates (under the `customConstraints` field):
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "replicalimit-policy.example.com"
+Spec:
+  failurePolicy: Fail
+  paramKind:
+    apiVersion: rules.example.com/v1
+    kind: ReplicaLimit
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["apps"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["deployments"]
+  customConstraints:
```

Comment:
IMO, these should either be called `matchPredicates`, or move the field into `matchConstraints` and just call it `predicates`.

`customConstraints` is not obvious what it does from the name alone, and could easily be confused with `validations`.

Reply:
I'm okay with either (thought `predicates` would maybe be too jargon-y for some).

I probably wouldn't call it `matchPredicates` since that is a bit too prescriptive (there may be other uses for predicates we haven't thought of).

We should avoid putting anything with CEL in the `matchConstraints` field (at least at first) as that would mean adding CEL to the parameter resource, which means it can no longer be treated as pure data.

I'd like to hear others' thoughts on naming so we can bikeshed on this thread and hopefully only make one commit.

As it stands, I'd rename it to `predicates`.

Reply:
Oh, I like that. It's a bit less jargony and I think it gets the point across.

Reply:
I'm OK with `preconditions` the only other field name I had to offer was `matchExpressions` or something prefixed with `match` to hint that the logic is applied in the matching phase.

Reply:
I'd also accept `matchConditions`. Let's agree on something across the two KEPs though.

Reply:
Ooh, I like `matchConditions`, WDTY @maxsmythe?

Reply:
Sounds great to me!

Reply:
Changed text to matchConditions